src/group_theory/divisible.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import group_theory.subgroup.pointwise
+import data.real.basic
+
+/-!
+# Divisible Group
+
+An abelian group `A` is said to be divisible iff `nA = A` for all `n ≠ 0`.
+
+## Main results
+
+* `add_comm_group.divisbible_of_linear_solvable` : If in a group `A`, `n • x = y` is solveable
+  whenever `n ≠ 0`, then `A` is divisible.
+* `add_comm_group.divisble_rat` : `ℚ` is a divisible group.
+* `add_comm_group.divisible_real` : `ℝ` is a divisible group.
+* `add_comm_group.divisible_of_product_divisible` : Any product of divisble group is divisible.
+
+TODO: Show that divisibility implies injectivity in the category of `AddCommGroup`.
+-/
+
+
+namespace add_comm_group
+
+open_locale pointwise
+
+variables (A : Type*) [add_comm_group A]
+
+/--
+A divisible group `A` is an abelian group such that `nA = A` for all `n ≠ 0`.
+-/
+class divisible : Prop :=
+(div : ∀ (n : ℤ), n ≠ 0 → n • (⊤ : add_subgroup A) = ⊤)
```

Comment:
Am I right in saying this could also be expressed as
```suggestion
(div : ∀ (n : ℤ), n ≠ 0 → surjective ((•) n : A → A))
```

Reply:
Yes, that is exactly `divisible_of_element_divisible`. I have changed to this as definition instead